R/zchunk_LA1322.Fert_china.R

```diff
@@ -0,0 +1,137 @@
+    L101.NBS_use_all_Mtce %>%
+      filter(EBProcess == "Agriculture", EBMaterial == "Coal Raw") %>%
+      # First zero out NAs in years where some values are NA but not all
+      group_by(province, year) %>%
+      mutate(value = replace(value, is.na(value) & sum(value, na.rm = T) != 0, 0)) %>%
+      ungroup ->
+      L1322.in_Mtce_province_Fert
+
+    L1322.in_Mtce_province_Fert %>%
+      group_by(province, sector, fuel) %>%
+      # use approx_fun rule = 2 to fill out data in years where the entire province is NA
+      mutate(value = approx_fun(year, value, rule = 2)) %>%
+      ungroup %>%
+      # When there is one data point, approx_fun replaced it with NA, but what we needed is to replace all NAs with that data.
+      left_join(L1322.in_Mtce_province_Fert %>% rename(org = value),
+                by = c("province", "fuel", "sector", "year", "EBProcess", "EBMaterial")) %>%
+      group_by(sector, fuel, province) %>%
+      # Only under the occasion when there is one data point in the group,
+      # approx_fun replaces the one value with NA.
+      # The original gcam_interp function replaces all missing values in that group with the one value when rule = 2.
+      mutate(value = replace(value, is.na(value) & sum(org[is.na(value)], na.rm = T) != 0,
+                             sum(org[is.na(value)], na.rm = T))) %>%
+      select(-org) %>%
+      ungroup ->
+      L1322.in_Mtce_province_Fert
+
+    # Convert to province precentages
+
+    L1322.in_Mtce_province_Fert %>%
+      group_by(sector, fuel, year) %>%
+      mutate(multiplier = value / sum(value)) %>%
+      ungroup %>%
+      group_by(province, fuel) %>%
+      mutate(multiplier = approx_fun(year, multiplier, rule = 2), sector = "N fertilizer") %>%
+      ungroup() %>%
+      select(-value, -EBProcess, -EBMaterial, -fuel) %>%
+      # Use the same shares for all fuels
+      repeat_add_columns(tibble(fuel = unique(L1322.Fert_Prod_MtN_R_F_Y$fuel))) ->
+      L1322.in_pct_province_Fert_repF
+
+    # Approtion national production to provinces
+    L1322.in_pct_province_Fert_repF %>%
+      left_join_error_no_match(filter(L1322.Fert_Prod_MtN_R_F_Y, GCAM_region_ID == gcamchina.REGION_ID), by = c("fuel", "sector", "year")) %>%
+      mutate(value = value * multiplier) %>% # Multiplying the national amount with the province share
+      select(province, sector, year, value, fuel) ->
+      L1322.out_Mt_province_Fert_Yh
+
+
+    # Assuming all provinces have the same IO coefficients
+    L1322.IO_R_Fert_F_Yh %>%
+      filter(GCAM_region_ID == gcamchina.REGION_ID) %>%
+      repeat_add_columns(tibble(province = gcamchina.PROVINCES)) %>%
+      filter(province != 'XZ') %>%
+      select(province, sector, fuel, year, value) ->
+      L1322.IO_GJkg_province_Fert_F_Yh
+
+    L1322.IO_GJkg_province_Fert_F_Yh %>%
+      left_join_error_no_match(L1322.out_Mt_province_Fert_Yh %>% rename(org = value), by = c("province", "year", "sector", "fuel")) %>%
```

Comment:
rename `value` as `output` or something meaningful with the data. Also make comments above line 103: calculate input by multiplying output with input-output coefficients